File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int gletters(string t);
int gsentences(string t);
int gwords(string t);
int main(void)
{
    string text = get_string("Text: ");
    int letters = gletters(text);
    int sentences = gsentences(text);
    int words = gwords(text);
    double L = letters / (float) words * 100;
    double S = sentences / (float) words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        if (index >= 1 && index <= 16)
        {
            printf("Grade %i\n", index);
        }
        else
        {
            printf("Grade 16+\n");
        }
    }
}
int gletters(string t)
{
    int letters = 0;
    for (int i = 0; i < strlen(t); i++)
    {
        if (isalpha(t[i]) != 0)
        {
            letters++;
        }
    }

    return letters;
}
int gwords(string t)
{
    int words = 1;
    for (int k = 0; k < strlen(t); k++)
    {
        if (t[k] == ' ')
        {
            words++;
        }
    }

    return words;
}
int gsentences(string t)
{
    int sentences = 0;
    for (int j = 0; j < strlen(t); j++)
    {
        if ((t[j] == '!') || (t[j] == '.') || (t[j] == '?'))
        {
            sentences++;
        }
    }

    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
sentences = 0
words = 1
for c in text:
    if c.isalpha():
        letters += 1
    elif c.isspace():
        words += 1
    elif c in ["!", ".", "?"]:
        sentences += 1
l = letters / words * 100
s = sentences / words * 100
index = round(0.0588 * l - 0.296 * s - 15.8)
if index < 1:
    print("Before Grade 1")
else:
    if index >= 1 and index <= 16:
        print(f"Grade {index}")
    else:
        print("Grade 16+")
